File: src/main/java/com/unicolour/joyspace/config/BeanConfiguration.java
package com.unicolour.joyspace.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.TaskScheduler;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.security.crypto.password.StandardPasswordEncoder;
import org.springframework.web.client.RestTemplate;

import java.security.SecureRandom;

@Configuration
@EnableScheduling
public class BeanConfiguration {
    @Bean
    public PasswordEncoder passwordEncoder() {
        return new StandardPasswordEncoder("R#0P=V0ut@Bx.Pd7");
    }

    @Bean
    public TaskScheduler taskScheduler() {
        return new ThreadPoolTaskScheduler();
    }

    @Bean
    public ObjectMapper objectMapper() {
        return new ObjectMapper();
    }

    @Bean
    public RestTemplate restTemplate() {
        return new RestTemplate();
    }

    @Bean
    public SecureRandom secureRandom() {
        return new SecureRandom();
    }
}

File: src/main/java/com/unicolour/joyspace/config/BeanConfiguration.kt
package com.unicolour.joyspace.config

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.TaskScheduler
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.security.crypto.password.StandardPasswordEncoder
import org.springframework.web.client.RestTemplate

import java.security.SecureRandom

@Configuration
@EnableScheduling
open class BeanConfiguration {
    @Bean
    open fun passwordEncoder(): PasswordEncoder = StandardPasswordEncoder("R#0P=V0ut@Bx.Pd7")

    @Bean
    open fun taskScheduler(): TaskScheduler = ThreadPoolTaskScheduler()

    @Bean
    open fun objectMapper() = ObjectMapper()

    @Bean
    open fun restTemplate() = RestTemplate()

    @Bean
    open fun secureRandom() = SecureRandom()
}
